pf/internal/check/check_test.go

```diff
@@ -96,6 +155,97 @@ func TestSensitiveIDWithOverride(t *testing.T) {
 	})
 }
 
+func TestInvalidInputID(t *testing.T) {
+	tests := []struct {
+		name            string
+		idSchema        idSchema
+		isInputProperty bool
+	}{
+		{name: "Required", idSchema: idSchema{required: true, optional: false, computed: false}, isInputProperty: true},
+		{name: "Optional", idSchema: idSchema{required: false, optional: true, computed: false}, isInputProperty: true},
+		{name: "Computed", idSchema: idSchema{required: false, optional: false, computed: true}, isInputProperty: false},
+		{
+			name:            "Optional+Computed",
+			idSchema:        idSchema{required: false, optional: true, computed: true},
+			isInputProperty: true,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name+" no overrides", func(t *testing.T) {
+			stderr, err := test(t, tfbridge.ProviderInfo{
+				P: pfbridge.ShimProvider(testProvider{withID: &tc.idSchema}),
+			})
+			if tc.isInputProperty {
+				assert.Error(t, err)
+				autogold.Expect(`error: Resource test_res has a problem: an "id" input attribute is not allowed. `+
+					`To map this resource specify SchemaInfo.Name and ResourceInfo.ComputeID
+`).Equal(t, stderr)
+			} else {
+				assert.Empty(t, stderr)
+				assert.NoError(t, err)
+			}
+		})
+
+		// "id" is a required output property so if we remap "id" -> "otherId" now we
+		// don't have an "id" output. We must create one by using `ComputeID`
+		t.Run(tc.name+" overrides with Name and missing ComputeID", func(t *testing.T) {
+			stderr, err := test(t, tfbridge.ProviderInfo{
+				P: pfbridge.ShimProvider(testProvider{withID: &tc.idSchema}),
+				Resources: map[string]*tfbridge.ResourceInfo{
+					"test_res": {Fields: map[string]*tfbridge.SchemaInfo{
+						"id": {Name: "otherId"},
+					}},
+				},
+			})
+			assert.Error(t, err)
+			autogold.Expect(`error: Resource test_res has a problem: an "id" attribute with SchemaInfo.Name `+
+				`must also specify ResourceInfo.ComputeID
+`).Equal(t, stderr)
+		})
+
+		// Providing `ComputeID` handles remapping the output "id" to a different property, but
```

Comment:
This looks right. I think with some elbow grease we could make the bridge to work with inputty IDs in this case but that can wait until some practical use case requires it.